cmd/launcher/svc_windows.go

```diff
@@ -231,6 +227,13 @@ func (w *winSvc) Execute(args []string, r <-chan svc.ChangeRequest, changes chan
 					"change_request", fmt.Sprintf("%+v", c),
 				)
 			}
+		case <-runLauncherResults:
+			w.systemSlogger.Log(ctx, slog.LevelInfo,
+				"shutting down service after runLauncher exited",
+			)
+			// We don't want to tell the service manager that we've stopped on purpose,
+			// so that the service manager will restart launcher correctly.
+			return false, uint32(windows.ERROR_EXCEPTION_IN_SERVICE)
```

Comment:
Love the comment. What let you to chose `ERROR_SERVICE_SPECIFIC_ERROR`?

Reply:
I did `windows.ERROR_EXCEPTION_IN_SERVICE` because that's what the windows service code defaults to (https://github.com/golang/sys/blob/master/windows/svc/service.go#L265). I took a look thru the [options](https://learn.microsoft.com/en-us/windows/win32/debug/system-error-codes--1000-1299-) and also considered `ERROR_PROCESS_ABORTED` and `ERROR_SERVICE_SPECIFIC_ERROR`. I didn't do a service-specific error code because we don't see or handle that error code (even if I set `ssec` to true) so it didn't seem worth defining launcher-specific codes. I figured the precedent was good enough of a reason to go with `ERROR_EXCEPTION_IN_SERVICE`.

I'll update the comment to include this context!